bundles/org.openhab.binding.dwdpollenflug/src/main/java/org/openhab/binding/dwdpollenflug/internal/handler/DWDPollenflugRegionHandler.java

```diff
@@ -0,0 +1,127 @@
+
+    private final Logger logger = LoggerFactory.getLogger(DWDPollenflugRegionHandler.class);
+
+    private DWDPollenflugRegionConfiguration thingConfig = new DWDPollenflugRegionConfiguration();
+
+    public DWDPollenflugRegionHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing DWD Pollenflug region handler");
+        thingConfig = getConfigAs(DWDPollenflugRegionConfiguration.class);
+
+        if (thingConfig.isValid()) {
+            DWDPollenflugBridgeHandler handler = syncToBridge();
+            if (handler == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Bridge handler missing");
+            } else {
+                handler.registerRegionListener(this);
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    private synchronized @Nullable DWDPollenflugBridgeHandler syncToBridge() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof DWDPollenflugBridgeHandler) {
+                DWDPollenflugBridgeHandler bridgeHandler = (DWDPollenflugBridgeHandler) handler;
+                return bridgeHandler;
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("DWDPollenflug region handler disposes. Unregistering listener.");
+        DWDPollenflugBridgeHandler bridgeHandler = syncToBridge();
+        if (bridgeHandler != null) {
+            bridgeHandler.unregisterRegionListener(this);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refresh();
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        refresh();
+    }
+
+    private void refresh() {
+        DWDPollenflugBridgeHandler handler = syncToBridge();
+        if (handler != null) {
+            DWDPollenflug pollenflug = handler.getPollenflug();
+            if (pollenflug != null) {
+                notifyOnUpdate(pollenflug);
+            }
+        }
+    }
+
+    @Override
+    public void notifyOnUpdate(DWDPollenflug pollenflug) {
+        DWDRegion region = pollenflug.getRegion(thingConfig.regionID);
+        if (region == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Region not found");
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        updateProperties(region.getProperties());
+
+        region.getChannels().forEach((channelID, value) -> {
+            logger.debug("Updating channel {} to {}", channelID, value);
+            updateState(channelID, value);
+        });

```

Comment:
```suggestion
        region.getChannels().forEach(this::updateState);
```

item-state-changes are logged by default, so if the channel is linked, this will be already logged.

Reply:
Changed